interfaces/C/ssids.f90

```diff
@@ -47,6 +48,12 @@ module spral_ssids_ciface
      character(C_CHAR) :: unused(76)
   end type spral_ssids_inform
 
+  type, bind(C) :: spral_numa_region
```

Comment:
Could `c_numa_region` from `hw_topology.f90` be renamed and used instead of redefining the same thing here?